model-api-gen/src/main/kotlin/org/modelix/metamodel/generator/MetaModelGenerator.kt

```diff
@@ -317,7 +317,7 @@ class MetaModelGenerator(val outputDir: Path, val nameConfig: NameConfig = NameC
                         if (feature.type is EnumPropertyType) {
                             if (serializer == MandatoryEnumSerializer::class.asTypeName()) {
                                 propBuilder.initializer(
-                                    """newProperty(%S, %S, %T { if (it != null) %T.valueOf(it) else %T.defaultValue() }, ${feature.optional})""",
+                                    """newProperty(%S, %S, %T { if (it != null) %T.valueOf(it.substringAfterLast('/')) else %T.defaultValue() }, ${feature.optional})""",
```

Comment:
This should go into MandatoryEnumSerializer/OptionalEnumSerializer. Then don't use the name, but export the ID of the enum member from MPS and use the part before the slash for deserialization.

The implementation of enum serialization/deserialization in MPS can be found here: https://github.com/JetBrains/MPS/blob/346e4440f6b05dabdc34f1be67849bdc942761b4/core/kernel/source/jetbrains/mps/smodel/adapter/structure/types/SEnumerationAdapter.java#L113-L181